### PR TITLE
Enduring Luminescence

### PR DESCRIPTION
**Pull request by rp4rk**
Implements the Enduring Luminescence trait for Discipline Priests.

### CODE REVIEW

File: src/parser/priest/discipline/modules/azeritetraits/EnduringLuminescence.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  }
+
+  on_byPlayer_heal(event) {
+    this.handlePWRBonus(event);
+    this.handleAtonementBonus(event);
+  }
+
+  handlePWRBonus(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.POWER_WORD_RADIANCE.id) return;
```
Comment (joshinat0r) [inline src/parser/priest/discipline/modules/azeritetraits/EnduringLuminescence.js]:
```
if () {
    return;
}
```
just to keep it consistent across the code-base. I personally prefer it this way aswell, but 🤷‍♂️

File: src/parser/priest/discipline/modules/azeritetraits/EnduringLuminescence.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    atonementApplicationSource: AtonementApplicationSource,
+    statTracker: StatTracker,
+    atonement: Atonement,
+  };
+
+  constructor(...args) {
+    super(...args);
+
+    this.ranks =
+      this.selectedCombatant.traitRanks(SPELLS.ENDURING_LUMINESCENCE.id) || [];
```
Comment (joshinat0r) [inline src/parser/priest/discipline/modules/azeritetraits/EnduringLuminescence.js]: I think Zero had a few weeks ago an argument about max-line length stuff and making too much use of the enter-key.
This should be one line

File: src/parser/priest/discipline/modules/azeritetraits/EnduringLuminescence.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    this.bonusHealing =
+      this.ranks
+        .map(rank =>
```
Comment (joshinat0r) [inline src/parser/priest/discipline/modules/azeritetraits/EnduringLuminescence.js]: Same as above, I think we usually format it this way.
```
this.bonusHealing = this.ranks
                     .map(rank => calculateAzeriteEffects(SPELLS.ENDURING_LUMINESCENCE.id, rank))
                     .reduce((total, bonus) => total + bonus, 0) * HEALING_MULTIPLIER;
```

File: src/parser/priest/discipline/modules/azeritetraits/EnduringLuminescence.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+    this.castBonusHealing += eventHealing;
+  }
+
+  handleAtonementBonus(event) {
+    // Atonement portion
+    const target = this.atonement.currentAtonementTargets.find(
+      id => id.target === event.targetID
+    );
+    if (!target) return;
```
Comment (joshinat0r) [inline src/parser/priest/discipline/modules/azeritetraits/EnduringLuminescence.js]: Return

File: src/parser/priest/discipline/modules/azeritetraits/EnduringLuminescence.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    if (!target.isEnduringEmpowered) return;
```
Comment (joshinat0r) [inline src/parser/priest/discipline/modules/azeritetraits/EnduringLuminescence.js]: Return

File: src/parser/priest/discipline/modules/azeritetraits/EnduringLuminescence.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    if (
+      event.timestamp <
+      target.atonementExpirationTimestamp - EnduringLuminescence.bonusDuration
+    )
+      {return;}
```
Comment (joshinat0r) [inline src/parser/priest/discipline/modules/azeritetraits/EnduringLuminescence.js]:
```
if (
      event.timestamp <
      target.atonementExpirationTimestamp - EnduringLuminescence.bonusDuration
    ) {
    return;
}
```

File: src/parser/priest/discipline/modules/azeritetraits/EnduringLuminescence.js [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+  }
+
+  statistic() {
+    return (
+      <DualStatisticBox
+        footer="PWR HPS is from the healing bonus."
+        category={STATISTIC_CATEGORY.AZERITE_POWERS}
+        position={STATISTIC_ORDER.OPTIONAL()}
+        icon={
+          (
```
Comment (joshinat0r) [inline src/parser/priest/discipline/modules/azeritetraits/EnduringLuminescence.js]: The `(`-bracket isn't necessary and the spellIcon should be in the same line as the icon-prop

**Review by joshinat0r — CHANGES_REQUESTED** [review]
Just some nitpicking on the formatting, the code itself looks good.

File: src/parser/priest/discipline/modules/azeritetraits/EnduringLuminescence.js (hunk 7 above)
Comment (rp4rk) [reply src/parser/priest/discipline/modules/azeritetraits/EnduringLuminescence.js]: This breaks our linting unfortunately, I'll meet at the middle.

**Review by joshinat0r — APPROVED** [review]

**Comment by joshinat0r** [comment]
Travis failed because the StatTracker is now located in `src/parser/shared/modules/StatTracker`

File: src/parser/priest/discipline/modules/azeritetraits/EnduringLuminescence.js [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+    this.handleAtonementBonus(event);
+  }
+
+  handlePWRBonus(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.POWER_WORD_RADIANCE.id) {
+      return;
+    }
+
+    let eventHealing = this.bonusHealing;
```
Comment (Gebuz) [inline src/parser/priest/discipline/modules/azeritetraits/EnduringLuminescence.js]: In general traits are affected by secondary stats and multipliers (except primary stat).

**Review by joshinat0r — CHANGES_REQUESTED** [review]
StatTracker location

**Pull request closed without merging**